mikupad.html

```diff
@@ -1873,7 +2211,64 @@
 	return [value, updateState];
 }
 
+
+function serializeTemplateCollection(collection) {
+	const templatesArray = [...collection.templates.entries()].map(([id, {name, prefix, suffix}]) => ({id, name, prefix, suffix}));
+	return JSON.stringify({
+		templates: templatesArray,
+		nextId: collection.nextId,
+		selectedTemplateId: collection.selectedTemplateId,
+	});
+}
+
+function deserializeTemplateCollection(serialized) {
+	const parsed = JSON.parse(serialized);
+	const collection = new TemplateCollection();
+	collection.templates.clear();
+	collection.nextId = parsed.nextId;
+	collection.selectedTemplateId = parsed.selectedTemplateId;
+	parsed.templates.forEach(({id, name, prefix, suffix}) => {
+		collection.templates.set(id, {name, prefix, suffix});
+		collection.nextId = Math.max(collection.nextId, id + 1);
+	});
+	return collection;
+}
+
+function usePersistentTemplateCollection(name, initialState) {

```

Comment:
This whole function looks very much like a hack, I don't think this is a good way of doing it. This functionality should probably be in the `TemplateCollection`.